src/core_plugins/elasticsearch/lib/cluster.js

```diff
@@ -0,0 +1,133 @@
+import elasticsearch from 'elasticsearch';
+import { get, set, isEmpty, cloneDeep, pick } from 'lodash';
+import toPath from 'lodash/internal/toPath';
+import Boom from 'boom';
+
+import filterHeaders from './filter_headers';
+import { parseConfig } from './parse_config';
+
+export default class Cluster {
+  constructor(config) {
+    this._config = Object.assign({}, config);
+    this.errors = elasticsearch.errors;
+
+    this._client = this.createClient();
+    this._noAuthClient = this.createClient({ auth: false });
+
+    return this;
+  }
+
+  /**
+   * callWithRequest
```

Comment:
Whatever you decide is fine, but personally I love the argument specifications. :)  I just don't like repeating the name of the function.  I like it when input parameter types are explicitly defined because I find it difficult to discover since js is not strongly typed.  Additionally you can get some visual indication (depending on your IDE) that you are doing something wrong if you pass in a type that isn't expected.